src/java/org/apache/cassandra/db/guardrails/Guardrails.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.db.guardrails;
+
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.service.ClientState;
+
+import static java.lang.String.format;
+
+/**
+ * Entry point for Guardrails, storing the defined guardrails and provided a few global methods over them.
+ */
+public abstract class Guardrails
+{
+    public static final String CUSTOM_GUARDRAILS_CONFIG_PROVIDER_PROPERTY = "cassandra.custom_guardrails_config_provider_class";
+
+    public static final GuardrailsConfigProvider config = System.getProperty(CUSTOM_GUARDRAILS_CONFIG_PROVIDER_PROPERTY) == null
+                                                                 ? new GuardrailsConfigProvider.Default()
+                                                                 : GuardrailsConfigProvider.build(System.getProperty(CUSTOM_GUARDRAILS_CONFIG_PROVIDER_PROPERTY));
```

Comment:
I am cool with this if there is 0 support; aka can break in a minor.  Can we add a comment in the code to make this clear?

Reply:
I have added comments about it in both `GuardrailsConfigService` and `GuardrailsConfig`. I have also removed the setters from the `GuardrailsConfig` interface since we don't need them there and we can have a smaller API that way.